src/dpos-user.ts

```diff
@@ -99,46 +99,32 @@ export class DPOSUser {
   static async createEthSignMetamaskUserAsync(
     web3: Web3,
     dappchainEndpoint: string,
-    dappchainKey: string,
     chainId: string,
     gatewayAddress: string,
     loomAddress: string
   ): Promise<DPOSUser> {
     const wallet = getMetamaskSigner(web3.currentProvider)
 
-    const { client, address } = await createDefaultEthSignClientAsync(
-      dappchainKey,
-      dappchainEndpoint,
-      chainId,
-      wallet
-    )
+    const client = await createDefaultEthSignClientAsync(dappchainEndpoint, chainId, wallet)
 
     const ethAddress = await wallet.getAddress()
+    const callerAddress = new Address('eth', LocalAddress.fromHexString(ethAddress))
```

Comment:
createDefaultEthSignClientAsync already calls wallet.getAddress so you could just return the caller address from the create function instead.